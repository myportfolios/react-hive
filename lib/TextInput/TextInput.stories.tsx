import { fn } from "@storybook/test";
import { TextInput } from "./TextInput";

export const ActionsData = {
  handleUserInput: fn(),
};
export default {
  title: "Components/TextInput",
  component: TextInput,
  tags: ["autodocs"],
  args: {
    ...ActionsData,
  },
};

export const Small = {
  args: {
    size: "sm",
  },
};

export const Medium = {
  args: {
    size: "md",
  },
};

export const Large = {
  args: {
    size: "lg",
  },
};

export const ExtraLarge = {
  args: {
    size: "xl",
  },
};
