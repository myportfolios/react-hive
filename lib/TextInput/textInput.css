@tailwind base;
@tailwind components;
@tailwind utilities;

.text-input {
  @apply block w-56 rounded-md py-1.5 px-2 ring-1 ring-inset ring-primary focus:accent-accent1;
}
.variant-sm {
  @apply w-48;
}

.variant-md {
  @apply w-64;
}

.variant-lg {
  @apply w-80;
}
.variant-xl {
  @apply w-96;
}
/* .error {
  @apply text-red-500;
} */
