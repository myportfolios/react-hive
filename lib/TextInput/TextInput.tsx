import "./textInput.css";

interface TextInputProps {
  handleUserInput: (value: string) => void;
  onKeyboardEnter?: (value: string) => void;
  type: "text" | "password" | "email";
  value: string;
  styles?: string;
  style?: React.CSSProperties;
  size?: string;
  error?: string;
  isVertical?: boolean;
}
export function TextInput({
  handleUserInput,
  onKeyboardEnter,
  type = "text",
  value,
  styles = "",
  style = {},
  size = "sm",
  error = "Error",
  isVertical = false,
}: TextInputProps) {
  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    if (handleUserInput) handleUserInput(event.target.value); // Pass just the value
  };
  const handleKeyPress = (event: React.KeyboardEvent<HTMLInputElement>) => {
    if (event.key === "Enter" && onKeyboardEnter) {
      onKeyboardEnter(value);
    }
  };
  const textInputStyles = `text-input  variant-${size} ${styles}`;
  return (
    <div className={`flex ${isVertical ? "isVertical" : ""}`}>
      <input
        type={type}
        placeholder="placeholder"
        value={value}
        onChange={handleChange}
        onKeyDown={handleKeyPress}
        className={textInputStyles}
        style={style}
        role="textbox"
      />
      {error && <span className="error">{error}</span>}
    </div>
  );
}
