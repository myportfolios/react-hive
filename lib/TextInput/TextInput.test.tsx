import { render } from "@testing-library/react";
import { TextInput } from "./TextInput";

describe(`Component: ${TextInput.name}`, () => {
  it("should render", () => {
    const handleChange = () => {};
    const { container } = render(
      <TextInput
        value="This is TextInput Component"
        handleUserInput={handleChange}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
