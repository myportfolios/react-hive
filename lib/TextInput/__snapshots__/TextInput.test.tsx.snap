// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component: TextInput > should render 1`] = `
<div>
  <input
    class="text-input  variant-sm "
    placeholder="placeholder"
    type="text"
    value="This is TextInput Component"
  />
</div>
`;
